src/rules/no-confusing-set-timeout.ts

```diff
@@ -0,0 +1,81 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createRule, getNodeName, parseJestFnCall } from './utils';
+
+interface ErrorType {
+  messageId: 'globalSetTimeout' | 'multipleSetTimeouts' | 'orderSetTimeout';
+  node: TSESTree.Node;
+}
+
+function isJestSetTimeout(node: TSESTree.Node) {
+  return getNodeName(node) === 'jest.setTimeout';
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow using confusing setTimeout in test',
+      recommended: false,
+    },
+    messages: {
+      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope.',
+      multipleSetTimeouts:
+        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect.',
+      orderSetTimeout:
+        '`jest.setTimeout` should be placed before any other jest methods',
+    },
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const errors: ErrorType[] = [];
+    let callJestTimeout = 0;
+    let nonJestTimeout = 0;
+
+    return {
+      CallExpression(node) {
+        const scope = context.getScope();
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) return;
+
+        if (
```

Comment:
If `jestFnCall` returns non-null it means that we believe the call is something from Jest, even if the type is `unknown` - so we should be fine to just check if the call is to `setTimeout` or not; and given that I'd move all that logic into `isJsetSetTimeout` (i.e. have it take the `ParsedJestFnCall` and handle checking that the `type` is `jest`)